File: 7-kyu/drone-fly-by/python/solution.py
def fly_by(lamps, drone):
    drone_len = len(drone)
    lamps_len = len(lamps)
    return f'{"o" * ((drone_len - 1) if drone_len > lamps_len else drone_len)}{"x" * min(lamps_len - drone_len, lamps_len)}'

File: 7-kyu/drone-fly-by/javascript/solution.js
const flyBy = (lamps, drone) => {
  const lampsLen = lamps.length;
  const droneLen = drone.length;
  const oRepeat = droneLen > lampsLen ? droneLen - 1 : droneLen;
  const xRepeat = Math.max(Math.min(lampsLen - droneLen, lampsLen), 0);
  return `${'o'.repeat(oRepeat)}${'x'.repeat(xRepeat)}`;
};
